cmd/channel_account.go

```diff
@@ -0,0 +1,111 @@
+package cmd
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/stellar/go/clients/horizonclient"
+	"github.com/stellar/go/support/config"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/wallet-backend/cmd/utils"
+	"github.com/stellar/wallet-backend/internal/db"
+	"github.com/stellar/wallet-backend/internal/services"
+	"github.com/stellar/wallet-backend/internal/signing"
+	"github.com/stellar/wallet-backend/internal/signing/channelaccounts"
+)
+
+type channelAccountCmdConfigOptions struct {
+	DatabaseURL                   string
+	HorizonClientURL              string
+	NetworkPassphrase             string
+	BaseFee                       int
+	DistributionAccountPrivateKey string
+}
+
+type channelAccountCmd struct {
+	channelAccountService services.ChannelAccountService
+}
+
+func (c *channelAccountCmd) Command() *cobra.Command {
+	cfg := channelAccountCmdConfigOptions{}
+	cfgOpts := config.ConfigOptions{
+		utils.DatabaseURLOption(&cfg.DatabaseURL),
+		utils.NetworkPassphraseOption(&cfg.NetworkPassphrase),
+		utils.BaseFeeOption(&cfg.BaseFee),
+		utils.HorizonClientURLOption(&cfg.HorizonClientURL),
+		utils.DistributionAccountPrivateKeyOption(&cfg.DistributionAccountPrivateKey),
```

Comment:
👋 you should have a separate encryption hey just for encrypting channel accounts. You don't want the host to have to handle the distribution secret when they need to rotate the channel account encryption key.